Description: State how to change the cache location in error messages

This mostly affects build-time testing of reverse dependencies,
as Debian buildds have no home directory for the build user.

Author: Rebecca N. Palmer <rebecca_palmer@zoho.com>
Bug-Debian: https://bugs.debian.org/896220 https://bugs.debian.org/896421
Forwarded: no

--- a/theano/configdefaults.py
+++ b/theano/configdefaults.py
@@ -1849,8 +1849,8 @@ def filter_compiledir(path):
         if not os.access(path, os.R_OK | os.W_OK | os.X_OK):
             # If it exist we need read, write and listing access
             raise ValueError(
-                "compiledir '%s' exists but you don't have read, write"
-                " or listing permissions." % path)
+                "compiledir (cache directory) '%s' exists but you don't have read, write"
+                " and/or listing permissions.  If you wish to change the cache location, set THEANO_FLAGS=base_compiledir='/cachedir/to/use' in your environment." % path)
     else:
         try:
             os.makedirs(path, 0o770)  # read-write-execute for user and group
@@ -1859,8 +1859,8 @@ def filter_compiledir(path):
             # the same directory at the same time.
             if e.errno != errno.EEXIST:
                 raise ValueError(
-                    "Unable to create the compiledir directory"
-                    " '%s'. Check the permissions." % path)
+                    "Unable to create the compiledir (cache directory)"
+                    " '%s'. Check the permissions.  If you wish to change the cache location, set THEANO_FLAGS=base_compiledir='/cachedir/to/use' in your environment." % path)
 
     # PROBLEM: sometimes the initial approach based on
     # os.system('touch') returned -1 for an unknown reason; the
